### DISCUSSION TITLE
General question about seeding

### DISCUSSION THREAD

**Discussion by Razorfang** [rakshasa/rtorrent#1185] [Q&A]
Hey, I'm pretty new to rtorrent, so I want to clarify some information about seeding.
Torrents end up in the `download/` directory by default. My .rtorrent.rc file has this line to represent that:
```
directory.default.set = (cat, (cfg.basedir), "download/")
```

I usually move the finished file to another directory when they're completed. If I move a file out of the download directory, what additional configuration do I need to ensure I am seeding this file?

For example, if I keep all my downloaded files in a directory called `bulk/` with different subdirectories for different types of media, is there a way I can tell rtorrent to seed files from there instead of from `download/`? Also, would `bulk/` have to be a subdirectory of  the base directory?

**Comment by kannibalox** [comment]
In addition to moving the data, you'll also need to tell rtorrent that the location has changed. Here are some example steps (completely untested and might not work for your setup, the `rtxmlrpc` command can be found [here](https://github.com/kannibalox/pyrosimple#installation)).

```bash
hash=AAAABBBB00001111222233334444555566667777
rtxmlrpc d.close ${hash} # ^K in the UI
mv "$(rtxmlrpc d.directory_base ${hash})/$(rtxmlrpc d.name ${hash})" "$(rtxmlrpc cfg.basedir)/bulk/" # Move the data
rtxmlrpc d.directory.set ${hash} "$(rtxmlrpc cfg.basedir)/bulk/" # ^O in the UI, be sure to provide the full path
rtxmlrpc d.start ${hash} # ^S in the UI
```

You haven't described how thing are being moved on completion, but if you're using a script that you got from somewhere else combined with `event.download.finished`, it's possible all of this is already happening and there's nothing else you need to do.

> If I move a file out of the download directory, what additional configuration do I need to ensure I am seeding this file?

No additional configuration is needed to allow rtorrent to seed from non-`directory.default` locations.

> Also, would bulk/ have to be a subdirectory of the base directory?

There are no such restrictions, rTorrent can seed from any file/directory the user has access to.

If you have any other questions, providing your full rtorrent.rc as well as any associated scripts would be very helpful.
